scalyr_agent/agent_main.py

```diff
@@ -2152,11 +2145,31 @@ def stop(self):
         "the background until it is successful.  This is useful if the network is not immediately "
         "available when the agent starts.",
     )
+
+    if __scalyr__.INSTALL_TYPE == __scalyr__.InstallType.TARBALL_INSTALL:
+        parser.add_option(
+            "",
+            "--check-frozen-binary",
+            action="store_true",
+            dest="check_frozen_binary",
+            help="NOTE: This option is intednded to be used by internal package scripts and there is no need to use it "
+                 "manually.\n "
+                 "This command is presented only in the Scalyr agent tarball package and it is used to diagnose "
+                 "if the current system is able to execute the frozen binary or it has to fallback to the source code. "
+                 "Its only purpose is to exit successfully (with zero code) and, thereby, "
+                 "to signal that the frozen binary is able to run on the current system. If the command fails, "
+                 "then, for some reason, the current system is not able to execute the frozen binary."
+        )
+
     my_controller.add_options(parser)
 
     (options, args) = parser.parse_args()
     my_controller.consume_options(options)
 
+    if __scalyr__.INSTALL_TYPE == __scalyr__.InstallType.TARBALL_INSTALL:
```

Comment:
Should we add explicit ``else`` with with non-zero exit and print "useful" error message? I know we will already exit with non-zero even without the else clause because of the code below, but it may be confusing since "unrelated" error message will be printed.